File: ClassPrevious/class/01/Hw2/Hw2/OGLCamera.java
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a graduate level course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Advanced Computer Graphics and Data visualization
//           Rensselaer Polytechnic Institute
//  Date:    January 1998
//
//////////////////////////////////////////////////////////////////////////

import jogl.*;

/**
  * Camera class handles viewport information
  *
  */
public class OGLCamera extends Camera
{
	JoglCanvas gl;

	/** Called by the Renderer after added */
	public void setRenderer(Renderer aren) { 
		renderer = aren;
		gl = ((OGLRenderer)aren).gl;
	}

	/**  Method to rotate the camera along the X axis */
    public void rotateX(float angle)
	{
		gl.matrixMode(GL.MODELVIEW);
		gl.rotate(angle, 1, 0, 0);
	}
	/**  Method to rotate the camera along the Y axis */
    public void rotateY(float angle)
	{
		gl.matrixMode(GL.MODELVIEW);
		gl.rotate(angle, 0, 1, 0);
	}
	/**  Method to rotate the camera along the Z axis */
    public void rotateZ(float angle)
	{
		gl.matrixMode(GL.MODELVIEW);
		gl.rotate(angle, 0, 0, 1);
	}

	/**  Method to translate the camera */
    public void translate(float x, float y, float z)
	{
		gl.matrixMode(GL.PROJECTION);
		gl.translate(x, y, z);
	}

	/** (Hopefully) a method to zoom the camera */
    public void zoom(float amount)
	{
		gl.matrixMode(GL.PROJECTION);
		gl.translate(0, 0, amount);
	}

	/** Convenience function to make the camera set the viewpoint */
    public void init()
	{
		gl.matrixMode(GL.PROJECTION);
		gl.loadIdentity();
		gl.perspective(eyeAngle, 
				1.0F*width/height, 
				clipRange[0], clipRange[1]);
		gl.lookAt(from.x, from.y, from.z,
						to.x, to.y, to.z,
						up.x, up.y, up.z);
	}
}

File: WebGL/OGLRen/OGLRen/OGLCamera.js
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a college course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Computer Graphics 
//           Rensselaer Polytechnic Institute
//  Date:    Sept 2015
//
//////////////////////////////////////////////////////////////////////////

var OGLRen = OGLRen || {};

/**
  * Camera class is a holder of cells into single objects.
  */
OGLRen.OGLCamera = function(aren) {
    var gl;
    var renderer;
    var pMatrix = mat4.create();
    mat4.identity(pMatrix);
    
    /** Render the camera **/
    this.render = function () {
        this.init();
        renderer.setPerspectiveMatrix(pMatrix);
    }
    
    /** Called by the Renderer after added */
    this.setRenderer = function( aren) { 
		renderer = aren;
		gl = aren.getGL();
        this.init();
	}

	/**  Method to rotate the camera along the X axis */
    this.rotateX = function( angle)
	{
		mat4.rotateX(pMatrix, pMatrix, angle);
	}
	/**  Method to rotate the camera along the Y axis */
    this.rotateY = function( angle)
	{
		mat4.rotateY(pMatrix, pMatrix, angle);
	}
	/**  Method to rotate the camera along the Z axis */
    this.rotateZ = function( angle)
	{
		mat4.rotateZ(pMatrix, pMatrix, angle);
	}

	/**  Method to translate the camera */
    this.translate = function( val)
	{
		mat4.translate(pMatrix, pMatrix, val);
	}
    
     /** (Hopefully) a method to scale the camera */
    this.scale = function(vec) {
        mat4.scale(pMatrix, pMatrix, vec);
    };


	/** (Hopefully) a method to zoom the camera */
    this.zoom = function( amount)
	{
		mat4.translate(pMatrix, pMatrix, [0, 0, amount]);
	}

	/** Convenience function to make the camera set the viewpoint */
    this.init = function()
	{
        if (this.initLess) {
            var width = renderer.getWidth();
            var height = renderer.getHeight();
            
            //mat4.perspective(pMatrix, 45, width / height, 1, 2000.0);
            /*mat4.lookAt(pMatrix, [this.from.x, this.from.y, this.from.z],
                            [this.to.x, this.to.y, this.to.z],
                            [this.up.x, this.up.y, this.up.z]);*/
            this.initLess = false;
        }
	}
    
    this.setRenderer(aren);
}
OGLRen.OGLCamera.inheritsFrom(OGLRen.Camera);
